File: https:/github.com/rajesh-kayal-dev/Leetcode.git/70-ClimbingStairs/70-ClimbingStairs.java
// Last updated: 6/15/2026, 10:02:45 AM
class Solution {
    public int climbStairs(int n) {
        if(n <= 2){
            return n;
        }

        int first = 1;
        int second = 2;

        for(int i = 3; i <= n; i++){
            int current = first + second;

            first = second;
            second = current;
        }
        return second;
    }
}

File: https:/github.com/rajesh-kayal-dev/Leetcode.git/70-ClimbingStairs/70-ClimbingStairs.js
// Last updated: 7/25/2026, 10:59:34 AM
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    if(n === 0 || n === 1) return 1;

    let prev = 1;
    let next = 1;
    let current = 0;

    for(let i = 2; i <= n; i++){
        current = prev + next;
        next = prev;
        prev = current;
    }

    return current;
};